src/plugins/discover/public/plugin.ts

```diff
@@ -366,6 +372,18 @@ export class DiscoverPlugin
 
     this.initializeServices();
 
+    // Datasources registrations for index patterns datasource
+    const { dataSourceService, dataSourceFactory } = plugins.data.dataSources;
+    dataSourceFactory.registerDataSourceType(DEFAULT_DATASOURCE_TYPE, DefaultDslDataSource);
```

Comment:
Having an incorrect type for `registerDataSourceType` shows as an error here when registering `DefaultDslDataSource`